controllers/useraccount/console_usersettings.go

```diff
@@ -0,0 +1,205 @@
+package useraccount
+
+import (
+	"context"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	ConsoleUserSettingsIdentifier         = "console.openshift.io/user-settings"
+	ConsoleUserSettingsUID                = "console.openshift.io/user-settings-uid"
+	UserSettingNS                         = "openshift-console-user-settings"
+	ConsoleUserSettingsResourceNamePrefix = "user-settings-"
+)
+
+// deleteConfigMap deletes a ConfigMap associated with a user from console setting.
+// It first attempts to find the ConfigMap by name. If not found, it then looks for the ConfigMap by label.
+// If multiple ConfigMaps are found by label, all of them are deleted.
+// The function returns an error if any occurred.
+func deleteConfigMap(ctx context.Context, cl client.Client, userUID string) error {
+	name := ConsoleUserSettingsResourceNamePrefix + userUID
+	logger := log.FromContext(ctx)
+	logger.Info(fmt.Sprintf("deleting configmap with name %s", name))
+	var toDelete []client.Object
+
+	// Create a ConfigMap object type to use with getConsoleUserSettingObjectByName function
+	objectType := &corev1.ConfigMap{TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"}}
+
+	// Attempt to find the ConfigMap by name
+	configMap, err := getConsoleUserSettingObjectByName(ctx, cl, name, objectType)
+	if err != nil {
+		return err
+	}
+	if configMap == nil {
+		// could not find the CM by name, try finding the configmap by label
+		if configMapList, err := getConfigMapByLabel(ctx, cl, userUID); err != nil || len(configMapList) == 0 {
+			return err
+		} else if len(configMapList) > 0 {
+			// if the number of items retrieved are more than one, should delete all of them
+			for key := range configMapList {
+				toDelete = append(toDelete, &configMapList[key])
+			}
+		}
+	} else {
+		toDelete = append(toDelete, configMap)
+	}
+
+	if err := deleteResources(ctx, cl, toDelete); err != nil {
+		return err
+	}
+	return nil
```

Comment:
as mentioned in the other comments, you don't need to duplicate the same code  - you can work with the generic `client.Object` or `unstructured.Unstructured` and  `unstructured.UnstructuredList` types. 
The pattern of the names as well as the labels seems to be the same for all resources, so you could theoretically have one generic function that would be called for all three types, just passing different GVK.
Please let me know if I'm missing some small detail.

Reply:
resolving this comment, as this was taken care of with the other changes suggested and implemented in [cefe5f3](https://github.com/codeready-toolchain/member-operator/pull/568/commits/cefe5f3f81dc73d4c681651eff2cc546c3e1bc18)